service/bfg/bfg.go

```diff
@@ -162,6 +165,37 @@ func NewServer(cfg *Config) (*Server, error) {
 	return s, nil
 }
 
+func (s *Server) queueCheckForInvalidBlocks() {
+	select {
+	case s.checkForInvalidBlocks <- struct{}{}:
+	default:
+	}
+}
+
+func (s *Server) invalidBlockChecker(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-s.checkForInvalidBlocks:
+			heights, err := s.db.BtcBlocksHeightsWithNoChildren(ctx)
+			if err != nil {
+				log.Errorf("error trying to get heights for btc blocks: %s", err)
+				return
+			}
+
+			log.Infof("received %d heights with no children, will re-check", len(heights))
+			for _, height := range heights {
+				log.Infof("reprocessing block at height %d", height)
+				if err := s.processBitcoinBlock(ctx, height); err != nil {
+					log.Errorf("error processing bitcoin block: %s", err)
+				}
+			}
+
```

Comment:
nit: could we remove this extra newline?

Reply:
yup!